Source/WebKit/UIProcess/API/Swift/WebPage.swift

```diff
@@ -120,20 +125,125 @@ public class WebPage_v0 {
+        }
+
+        guard let convertedEncoding = CFStringConvertEncodingToIANACharSetName(cfEncoding) as? String else {
+            preconditionFailure("\(characterEncoding) is not a valid character encoding")
+        }
+
+        return backingWebView.load(data, mimeType: mimeType, characterEncodingName: convertedEncoding, baseURL: baseURL).map(NavigationID.init(_:))
+    }
+
     @discardableResult
     public func load(htmlString: String, baseURL: URL) -> NavigationID? {
         backingWebView.loadHTMLString(htmlString, baseURL: baseURL).map(NavigationID.init(_:))
     }
 
+    @discardableResult
+    public func load(_ request: URLRequest, allowingReadAccessTo readAccessURL: URL) -> NavigationID? {
+        // `WKWebView` annotates this method as returning non-nil, but it may return nil.
+
+        let navigation = backingWebView.loadFileRequest(request, allowingReadAccessTo: readAccessURL) as WKNavigation?
+        return navigation.map(NavigationID.init(_:))
+    }
+
+    @discardableResult
+    public func loadSimulatedRequest(_ request: URLRequest, response: URLResponse, responseData: Data) -> NavigationID? {
+        // `WKWebView` annotates this method as returning non-nil, but it may return nil.
+
+        let navigation = backingWebView.loadSimulatedRequest(request, response: response, responseData: responseData) as WKNavigation?
+        return navigation.map(NavigationID.init(_:))
+    }
+
+    @discardableResult
+    public func loadSimulatedRequest(_ request: URLRequest, responseHTML: String) -> NavigationID? {
+        // `WKWebView` annotates this method as returning non-nil, but it may return nil.
+
+        let navigation = backingWebView.loadSimulatedRequest(request, responseHTML: responseHTML) as WKNavigation?
+        return navigation.map(NavigationID.init(_:))
+    }
+
+    @discardableResult
+    public func reload(fromOrigin: Bool = false) -> NavigationID? {
+        let navigation = fromOrigin ? backingWebView.reloadFromOrigin() : backingWebView.reload()
+        return navigation.map(NavigationID.init(_:))
+    }
+
+    public func stopLoading() {
+        backingWebView.stopLoading()
+    }
+
+    public func callAsyncJavaScript(_ functionBody: String, arguments: [String : Any] = [:], in frame: FrameInfo? = nil, contentWorld: WKContentWorld? = nil) async throws -> Any? {
+        try await backingWebView.callAsyncJavaScript(functionBody, arguments: arguments, in: frame?.wrapped, contentWorld: contentWorld ?? .defaultClient)
+    }
+
+#if canImport(UIKit)
+    public func snapshot(configuration: WKSnapshotConfiguration? = nil) async throws -> UIImage {
+        try await withCheckedThrowingContinuation { continuation in
+            backingWebView.takeSnapshot(with: configuration) { image, error in
+                if let error {
+                    continuation.resume(throwing: error)
+                    return
+                }
+
+                guard let image else {
```

Comment:
This isn’t really any different than force unwrapping, just more verbose. Given that all the invariants that would trigger that are in our own code, it’s probably fine to just use `!`. Though, did we not produce an `async` overload of this on `WKWebView` that we could just use instead?